reasoner_pydantic/results.py

```diff
@@ -141,6 +150,25 @@ def parse_obj(obj):
         result.update(r)
         return result
 
+    def combine_analyses_by_resource_id(self):
+        combine = HashableMapping[str, Analysis]()
+        analyses = HashableSequence.parse_obj([analysis for analysis in self.analyses])
```

Comment:
Why do we need to basically copy the analyses here?

Reply:
HashableSets are iterable. However, we use the indexes to ensure the upper triangular aspect of the loop, which is also what the enumeration is for. Otherwise, we have to loop through the entire loop twice, instead of looping through only those elements that are beyond the "analysis" in the second loop.

Reply:
That makes sense. Next question is, why not just make analyses a sequence to begin with? Are sets more performant?

Reply:
Sets ensure we don't have equivalent analyses being inserted, without having to check for that. We never want any service creating an analysis block that has the exact same value for every field. For that same reason, sets are more performant for merging.